fiftyone/utils/eval/openimages.py

```diff
@@ -0,0 +1,699 @@
+"""
+Open Images-style detection evaluation.
+
+| Copyright 2017-2021, Voxel51, Inc.
+| `voxel51.com <https://voxel51.com/>`_
+|
+"""
+from collections import defaultdict
+import copy
+import logging
+
+import matplotlib.pyplot as plt
+import numpy as np
+import sklearn.metrics as skm
+
+import fiftyone.core.utils as fou
+
+from .detection import (
+    DetectionEvaluation,
+    DetectionEvaluationConfig,
+    DetectionResults,
+)
+
+
+logger = logging.getLogger(__name__)
+
+
+class OpenImagesEvaluationConfig(DetectionEvaluationConfig):
+    """Open Images-style evaluation config.
+
+    Args:
+        pred_field: the name of the field containing the predicted
+            :class:`fiftyone.core.labels.Detections` instances
+        gt_field: the name of the field containing the ground truth
+            :class:`fiftyone.core.labels.Detections` instances
+        iou (None): the IoU threshold to use to determine matches
+        classwise (None): whether to only match objects with the same class
+            label (True) or allow matches between classes (False)
+        iscrowd ("IsGroupOf"): the name of the crowd attribute
+        max_preds (None): the maximum number of predicted objects to evaluate
+            when computing mAP and PR curves. 
+        hierarchy (None): a dict containing a hierachy of classes for
+            evaluation following the structure 
+            ``{"LabelName": label, "Subcategory": [{...}, ...]}``
+        pos_lab_field (None): the name of the field containing 
```

Comment:
nit: it's generally good to avoid abbreviations unless they really save space. So `pred` and `gt` are pretty ubiquitous in our codebase, and `pos` and `neg` are easy to understand. But `lab` only saves two characters compared to `label`, and the latter is more clear, so I'd recommend `pos_label_field` and `neg_label_field`.

Also, I believe these fields need to be `Classification` label fields, right? Whenever you reference a specific type like that, it's helpful to explicitly mention the class type `:class:`fiftyone.core.labels.Classification` so that readers of the API docs online can clickthrough to learn about the label type.